viewing_party/party.py

```diff
@@ -0,0 +1,179 @@
+
+    # iterate through the movie dictionaries that friends have watched
+    # and add the titles to a the friends_watched_movie_titles list 
+    for i in range(0, len(user_data["friends"])):
+        for movie in user_data["friends"][i].get("watched"):
+            friends_watched_movie_titles.append(movie["title"])
+
+    # iterate through the movies that the user has watched.
+    # if the title of a movie the user has watched is not in 
+    # the friends_watched_movie_titles list add the movie dictionary 
+    # to the user_unique_watched list
+    for user_movie in user_data["watched"]:
+        if user_movie.get("title") not in friends_watched_movie_titles:
+            user_unique_watched.append(user_movie)
+
+    # return the list of movie dictionaries representing movies only
+    # the user has watched
+    return user_unique_watched
+
+def get_friends_unique_watched(user_data):
+    # create list to host dictionaries of movies that the user's friends 
+    # have watched
+    friends_unique_watched_list = []
+    
+    # create list of strings of movie titles that the user has watched
+    user_watched_movie_titles = []
+
+    # create list of strings of movie titles that a user's friend has watched
+    friend_watched_movie_titles = []
+
+    # iterate through the movie dictionaries that the user has watched
+    # and add the titles to a the user_watched_movie_titles list
+    for movie_dict in user_data["watched"]:
+        user_watched_movie_titles.append(movie_dict.get("title"))
+
+    # check that the user has friends
+    if user_data["friends"]:
+    # iterate through the movies that the user's friends have watched.
+    # if the title of a movie the user's friends have watched is not in 
+    # the user_watched_movie_titles list add the movie dictionary 
+    # to the friends_unique_watched list
+        for i in range(0, len(user_data["friends"])):
+            for movie_dict in user_data["friends"][i]["watched"]:
+                if movie_dict["title"] not in user_watched_movie_titles and movie_dict["title"] not in friend_watched_movie_titles:
+                    friends_unique_watched_list.append(movie_dict)
+                    friend_watched_movie_titles.append(movie_dict["title"])
+
+    # return the list of movie dictionaries representing movies only
+    # the user has watched
+    return friends_unique_watched_list
+
+def get_available_recs(user_data):
+    # create a list of movie dictionaries for movies that the user 
+    # has not seen but a friend has seen, and that the user has access 
+    # to aka the movie's host is in the user's subscriptions list 
+    movie_recs = []
+
+    # iterate through movies that only the user's friends have seen
+    # check if the user has access to the movie's host
+    # check if the movie has already been added to the rec list
+    # if passes both checks add the movie to the rec list
+    for movie_dict in get_friends_unique_watched(user_data):
```

Comment:
Great job reusing `get_friends_unique_watched()` ! This function is really neat.